File: go/chapter9/three_restaurants.go
package main

import "fmt"

type Restaurant struct {
	restaurantName string
	cuisineType    string
}

func (res Restaurant) describeRestaurant() {
	fmt.Printf("Name: %s\n", res.restaurantName)
	fmt.Printf("Cuisine Type: %s\n", res.cuisineType)
}

func (res Restaurant) openRestaurant() {
	fmt.Printf("The %s Restaurant is now open!\n", res.restaurantName)
}

func main() {
	res1 := Restaurant{restaurantName: "Sunny", cuisineType: "Classic"}
	res1.openRestaurant()
	res1.describeRestaurant()

	res2 := Restaurant{restaurantName: "Nineties", cuisineType: "VIP"}
	res2.openRestaurant()
	res2.describeRestaurant()

	res3 := Restaurant{restaurantName: "Tech", cuisineType: "Nerds"}
	res3.openRestaurant()
	res3.describeRestaurant()

}

File: python/chapter9/three_restaurants.py
class Restaurant:
    def __init__(self, restaurant_name: str, cuisine_type: str):
        self.restaurant_name = restaurant_name
        self.cuisine_type = cuisine_type

    def describe_restaurant(self):
        print(f"Restaurant Name: {self.restaurant_name}")
        print(f"Cuisine Type: {self.cuisine_type}")

    def open_restaurant(self):
        print(f"The {self.restaurant_name} Restaurant is now open!")


res1 = Restaurant("Sunny", "Classic")
res1.open_restaurant()
res1.describe_restaurant()

res2 = Restaurant("Nineties", "VIP")
res2.open_restaurant()
res2.describe_restaurant()

res3 = Restaurant("Tech", "Nerds")
res3.open_restaurant()
res3.describe_restaurant()
